packages/network-controller/src/NetworkController.ts

```diff
@@ -1354,12 +1353,14 @@ export class NetworkController extends BaseControllerV2<
 
     let autoManagedNetworkClient: AutoManagedNetworkClient<NetworkClientConfiguration>;
 
+    let networkClientId: NetworkClientId;
     if (isInfuraProviderConfig(providerConfig)) {
       const networkClientType = NetworkClientType.Infura;
-      const networkClientId = buildInfuraNetworkClientId(providerConfig);
+      networkClientId = buildInfuraNetworkClientId(providerConfig);
       const builtInNetworkClientRegistry =
         this.#autoManagedNetworkClientRegistry[networkClientType];
-      autoManagedNetworkClient = builtInNetworkClientRegistry[networkClientId];
+      autoManagedNetworkClient =
```

Comment:
Sure. You'll forgive me if I raise this again somewhere else — I'm not trying to be annoying, I'm just working hard to ensure that we don't add to the tech debt that already exists in this repo. I do think it'd be worth it to understand why TypeScript is doing what it's doing, but if you're pressed on time or you think this will go away later anyway, I get it.

Reply:
oh yeah 100% on the same page about the importance of not overlooking typing issues. What I think I forgot to say was that if we can't get this resolved in a relatively short amount of additional effort, I will turn this into a seperate issue so that it can be tracked as a sort of 'side-quest' to this particular effort.

Reply:
seems that making a union of string literal + generic string is not super well supported by TS: https://github.com/microsoft/TypeScript/issues/29729

I've tried the provided work arounds to no avail. Its pretty silly that we treat networkclientIds differently if its infura or not to begin with, and as such im not going to spend anymore time on this one.

Reply:
Okay no problem.

As for treating network client IDs differently depending on Infura or not, I agree, it's silly and painful and creates extra code. It's our old friend again, though — we "need" this split because in the end, we build separate types of network clients. That said... I have an idea on how to solve this.